internal/clients/http/azure/create_vm.go

```diff
@@ -0,0 +1,395 @@
+	}
+	logger.Trace().Msgf("Using network security group id=%s", *nsg.ID)
+
+	networkInterface, err := c.createNetworkInterface(ctx, location, resourceGroupName, subnet, publicIP, nsg, nicName)
+	if err != nil {
+		span.SetStatus(codes.Error, "cannot create network interface")
+		logger.Error().Err(err).Msgf("cannot create network interface")
+		return nil, err
+	}
+	logger.Trace().Msgf("Using network interface id=%s", *networkInterface.ID)
+
+	virtualMachine, err := c.createVirtualMachine(ctx, location, resourceGroupName, networkInterface, imageID, sshKeyBody, vmName, diskName)
+	if err != nil {
+		span.SetStatus(codes.Error, "cannot create virtual machine")
+		logger.Error().Err(err).Msgf("cannot create virtual machine")
+		return nil, err
+	}
+	logger.Debug().Msgf("Created virtual machine id=%s", *virtualMachine.ID)
+
+	return virtualMachine, nil
+}
+
+func (c *client) createResourceGroup(ctx context.Context, name string, location string) (*armresources.ResourceGroup, error) {
+	resourceGroupClient, err := c.newResourceGroupsClient(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	parameters := armresources.ResourceGroup{
+		Location: ptr.To(location),
+		Tags:     map[string]*string{"sample-rs-tag": ptr.To("sample-tag")}, // resource group update tags
+	}
+
+	resp, err := resourceGroupClient.CreateOrUpdate(ctx, name, parameters, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	return &resp.ResourceGroup, nil
+}
+
+func (c *client) createVirtualNetwork(ctx context.Context, resourceGroupName string, name string) (*armnetwork.VirtualNetwork, error) {
+	ctx, span := otel.Tracer(TraceName).Start(ctx, "createVirtualNetwork")
+	defer span.End()
+
+	vnetClient, err := c.newVirtualNetworksClient(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	parameters := armnetwork.VirtualNetwork{
+		Location: to.Ptr(location),
+		Properties: &armnetwork.VirtualNetworkPropertiesFormat{
+			AddressSpace: &armnetwork.AddressSpace{
+				AddressPrefixes: []*string{
+					to.Ptr("10.42.0.0/16"),
+				},
+			},
+			//Subnets: []*armnetwork.Subnet{
+			//	{
+			//		Name: to.Ptr(subnetName+"3"),
+			//		Properties: &armnetwork.SubnetPropertiesFormat{
+			//			AddressPrefix: to.Ptr("10.1.0.0/24"),
+			//		},
+			//	},
+			//},
+		},
+	}
+
+	pollerResponse, err := vnetClient.BeginCreateOrUpdate(ctx, resourceGroupName, name, parameters, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := pollerResponse.PollUntilDone(ctx, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	return &resp.VirtualNetwork, nil
+}
+
+func (c *client) createSubnets(ctx context.Context, resourceGroupName string, vnetName string, name string) (*armnetwork.Subnet, error) {
+	ctx, span := otel.Tracer(TraceName).Start(ctx, "createSubnets")
+	defer span.End()
+
+	subnetClient, err := c.newSubnetsClient(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	parameters := armnetwork.Subnet{
+		Properties: &armnetwork.SubnetPropertiesFormat{
+			AddressPrefix: to.Ptr("10.42.42.0/24"),
+		},
+	}
```

Comment:
This should be the whole address space, why are you throwing out 8 bits for nothing?

For example, if we would go with my "wild guess" `172.22.0.0/16` this should be also the subnet.

Reply:
Yeah I guess you're right, there is no reason yet to have multiple subnets.